### PR TITLE
[SMALLFIX] Fix a bug when UnregisteredChannelFired

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/netty/AbstractReadHandler.java [hunk 1]
```diff
@@ -169,7 +169,9 @@ private void validateReadRequest(Protocol.ReadRequest request) throws InvalidArg
   private void setError(Channel channel, Error error) {
     Preconditions.checkNotNull(error);
     try (LockResource lr = new LockResource(mLock)) {
-      if (mContext.getError() != null) {
+      // Note, network errors may be bubbling up through channelUnregistered to reach here before
+      // mContext is initialized.
+      if (mContext == null || mContext.getError() != null) {
```
Comment (yupeng9) [inline core/server/worker/src/main/java/alluxio/worker/netty/AbstractReadHandler.java]: There are other references of `mContext` in the read handler and write handler. Do you think we should add null check for them?

File: core/server/worker/src/test/java/alluxio/worker/netty/BlockReadHandlerTest.java [hunk 2]
```diff
@@ -79,8 +79,8 @@ public void readFailure() throws Exception {
     long fileSize = PACKET_SIZE * 10 + 1;
     populateInputFile(0, 0, fileSize - 1);
     mBlockReader.close();
-    mChannelNoException.writeInbound(buildReadRequest(0, fileSize));
-    Object response = waitForOneResponse(mChannelNoException);
+    mChannel.writeInbound(buildReadRequest(0, fileSize));
```
Comment (yupeng9) [inline core/server/worker/src/test/java/alluxio/worker/netty/BlockReadHandlerTest.java]: why is this changed?

Comment (apc999) [inline core/server/worker/src/test/java/alluxio/worker/netty/BlockReadHandlerTest.java]: reverted

File: core/server/worker/src/main/java/alluxio/worker/netty/AbstractReadHandler.java (hunk 1 above)
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/netty/AbstractReadHandler.java]: added

**Comment by apc999** [comment]
@yupeng9  PTAL

**Comment by yupeng9** [comment]
jenkins, test this please
